### ISSUE TITLE
GraphQL variables do not work when uid is nestee

### ISSUE THREAD

**Issue by benmisell** [dgraph-io/dgraph#2355]
- What version of Dgraph are you using?
Docker dgraph/dgraph:latest

- Have you tried reproducing the issue with latest release?
Yes, ran docker pull to make sure it was up to date

- What is the hardware spec (RAM, OS)?
Ubuntu Studio 17.10, 8GB RAM, Pentium dual core B960 @ 2.20GHz

- Steps to reproduce the issue (command/config used to run Dgraph).
Docker compose from docks.
This schema:
```
booking.start: dateTime .
booking.end: dateTime .
booking.hotel: uid @reverse .
booking.room: uid @reverse .
booking.user: uid @reverse .
```

This query:
```
query q($userID: uid){
  bookings(func: has(booking)) @cascade {
    uid
    booking.start
    booking.end
    booking.hotel {
      uid
    }
    booking.room {
      uid
    }
    booking.user @filter(uid($userID)) {
      uid
    }
  }
}
```

- Expected behaviour and actual result.
The userId filter should find the user with that id, but does not. It finds nothing. As a temporary workaround this works.
```
query q($userID: uid){
  var (func: uid($userID)) {
    u as uid
  }
  bookings(func: has(booking)) @cascade {
    uid
    booking.start
    booking.end
    booking.hotel {
      uid
    }
    booking.room {
      uid
    }
    booking.user @filter(uid(u)) {
      uid
    }
  }
}
```

**Comment by pawanrawal** [comment]
Try specifying the type as `string` and let us know if it still doesn't work.

```
query q($userID: string){
...
```

**Comment by benmisell** [comment]
Unfortunately that still doesn't work. I pasted in the wrong code there, that query errors. My problem is that nothing is returned.

**Comment by pawanrawal** [comment]
This is a duplicate of https://github.com/dgraph-io/dgraph/issues/2088. I am working on a fix now.

**Comment by pawanrawal** [comment]
This should be fixed with 4d6a6077343570518cbe247fa1fbf9320731b09b which would be part of the nightly release.
